idaes/config.py

```diff
@@ -113,7 +129,31 @@
 }
 """
 
-def new_idaes_config_block():
+cfg = None # the idaes ConfigBlock once it's created by new_idaes_config_block()
+
+
+class ConfigDictJSONEncoder(json.JSONEncoder):
+    """ This class can be used to encode the ConfigDict as json.  It is
+    sufficient for the IDAES config, but not ConfigDicts in general.
+    """
+    def default(self, obj):
+        print(super().default)
+        if isinstance(obj, pyomo.common.config.ConfigDict):
+            d = {}
+            for k, v in obj.items():
+                d[k] = self.default(v)
+            return d
```

Comment:
This is not really necessary.  You can always do `config.value()`, which returns the data structure with `ConfigDict` resolved to `dict`, `ConfigList` to `list, and `ConfigValue` to the contained value.  You would still need to map `set()` -> `list`, though.

We could also add a JSON serializer to the ConfigBlock system (analogous to the YAML template generator)

Reply:
Thanks.  I knew that much be there, but I didn't immediately figure out how to do it.  I'll update.

I think a json serialized would be a nice addition if you have time.